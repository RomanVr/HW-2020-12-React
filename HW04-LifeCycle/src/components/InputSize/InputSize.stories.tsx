import React from "react";
import { action } from "@storybook/addon-actions";
import { text, withKnobs } from "@storybook/addon-knobs";
import { InputSize } from "..";

export default {
  title: "InputSize",
  Component: InputSize,
  decorators: [withKnobs],
};

export const InputSizeStory: React.FC = () => {
  return (
    <InputSize
      size={text("filled with", "size")}
      handleChangeSize={action("ChangeSize")}
    />
  );
};
