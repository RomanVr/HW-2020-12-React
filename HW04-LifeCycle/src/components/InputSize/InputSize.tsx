import React from "react";

interface InputSizeProp {
  size: string;
  handleChangeSize: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const stylesLabel = {
  display: "block-inline",
  background: "gray",
  border: "1px solid black",
  borderRadius: "5px",
  height: "30px",
};

const stylesInput = {
  background: "white",
  border: "1px solid grey",
  borderRadius: "5px",
  height: "30px",
};

export const InputSize: React.FC<InputSizeProp> = ({
  size,
  handleChangeSize,
}) => (
  <>
    <label style={stylesLabel}>Введите размер поля:</label>
    <br></br>
    <input
      style={stylesInput}
      type="text"
      value={size}
      onChange={handleChangeSize}
      placeholder="Enter size"
    ></input>
  </>
);
