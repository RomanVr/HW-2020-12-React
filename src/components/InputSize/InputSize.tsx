import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/react";

export interface InputSizeProps {
  size: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  placeHolder: string;
  label: string;
}

const stylesLabel = css`
  /* display: "block-inline"; */
  background: gray;
  border: 1px solid black;
  border-radius: 5px;
  height: 30px;
`;
const LabelInput = styled.label`
  ${stylesLabel}
`;

const stylesInput = css`
  /* display: "block-inline"; */
  flex: auto;
  white-space: pre;
  background: white;
  border: 1px solid grey;
  border-radius: 5px;
  height: 10px;
  text-align: center;
  padding: 5px;
  min-width: 30px;
`;
interface Props {
  value: number | string;
}
const stylesWidthInput = (props: Props) => css`
  width: ${String(props.value).length * 8}px;
`;
const InputItem = styled.input`
  ${stylesInput};
  ${stylesWidthInput}
`;

export const InputSize: React.FC<InputSizeProps> = ({
  size,
  onChange,
  placeHolder,
  label,
}) => (
  <>
    <LabelInput>
      {label}
      <InputItem
        data-testid="inputSize"
        type="text"
        value={size}
        onChange={onChange}
        placeholder={placeHolder}
      ></InputItem>
    </LabelInput>
  </>
);
