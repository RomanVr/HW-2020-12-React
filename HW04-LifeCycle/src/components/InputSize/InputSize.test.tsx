import React from "react";
import renderer from "react-test-renderer";
import { InputSize } from "./InputSize";

describe("InputSize", () => {
  it("Render input for epmty value", () => {
    expect(
      renderer
        .create(<InputSize size="" handleChangeSize={jest.fn()} />)
        .toJSON()
    ).toMatchSnapshot();
  });
});
