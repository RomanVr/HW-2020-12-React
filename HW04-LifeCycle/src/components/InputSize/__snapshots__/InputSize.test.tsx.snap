// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSize Render input for epmty value 1`] = `
Array [
  <label
    style={
      Object {
        "background": "gray",
        "border": "1px solid black",
        "borderRadius": "5px",
        "display": "block-inline",
        "height": "30px",
      }
    }
  >
    Введите размер поля:
  </label>,
  <br />,
  <input
    onChange={[MockFunction]}
    placeholder="Enter size"
    style={
      Object {
        "background": "white",
        "border": "1px solid grey",
        "borderRadius": "5px",
        "height": "30px",
      }
    }
    type="text"
    value=""
  />,
]
`;
